frontend/src/components/buttons/dropdowns/SourceSelect.tsx

```diff
@@ -0,0 +1,52 @@
+import React, { ReactElement } from "react";
+import { SourcePropsT } from "../../../types";
+import { Button } from "../Button";
+import { SourceMenuSvg } from "../../icons/SourceMenuSvg";
+
+import "../../../styles/Button.css";
+import "../../../styles/SourceSelect.css";
+
+interface Props {
+  sources: SourcePropsT[];
+}
+
+export function SourceSelect({ sources }: Props): ReactElement {
+  const [selectedSource, setSelectedSource] = React.useState(
+    {} as SourcePropsT
+  );
```

Comment:
Let's store only the `id` in here, that way if `sources` from props updates, we don't get out of sync.

```suggestion
  const [selectedSourceId, setSelectedSourceId] = React.useState(undefined);
```

---

Extra for experts (and totally unnecessary, but interesting): I watched a talk a while ago about better data structures for UI interactions, and one of the data structures mentioned was a `SelectionList`. You can see an implementation [in Elm here](https://package.elm-lang.org/packages/TheSeamau5/selection-list/latest/SelectionList). The idea is that you can never get inconsistent data, which is otherwise easier with simpler data structures.

Reply:
I think what you've done now is fine, we can always just change if needed. I think it might be better in the end to think about what we want the `sources` prop to convey. For me, a list implies that the data has an implied order, whereas an object means the data has no implied order, and can be rendered in any order.

Reply:
That's a fair point. Going to revert my changes to keep `sources` as a list for the time being. We should discuss the application state on the whole offline to keep in sync on the expected shape as more components get made.